src/jit/lowerxarch.cpp

```diff
@@ -2191,8 +2191,26 @@ void Lowering::TreeNodeInfoInitModDiv(GenTree* tree)
         info->setDstCandidates(l, RBM_RAX);
     }
 
-    // If possible would like to have op1 in RAX to avoid a register move
-    op1->gtLsraInfo.setSrcCandidates(l, RBM_RAX);
+#ifdef _TARGET_X86_
+    if (op1->OperGet() == GT_LONG)
+    {
+        GenTree* loVal = op1->gtGetOp1();
+        GenTree* hiVal = op1->gtGetOp2();
+        
+        // Src count is actually 3, so increment.
+        info->srcCount++;
```

Comment:
Minor: It is good idea to assert that op2 is of TYP_INT here and hence the reason for srcCount being 3.
assert(op2->IsCnsIntOrI());